packages/ketcher-macromolecules/src/Editor.tsx

```diff
@@ -239,13 +242,32 @@ function Editor({ theme, togglerComponent }: EditorProps) {
                 sequenceNode.rnaBase.monomerItem,
               ];
 
+        const existingPreset = presets.find((preset) => {
+          const presetMonomers = [preset.sugar, preset.base, preset.phosphate];
+          return monomers.every((monomer, index) => {
+            return monomer?.props.Name === presetMonomers[index]?.props.Name;
+          });
+        });
+
+        let position: PresetPosition;
+        if (sequenceNode instanceof Nucleoside) {
+          position = 'chainEnd';
+        } else if (sequenceNode.sugar.attachmentPointsToBonds.R1 === null) {
```

Comment:
I would suggest use sequenceNode.firstMonomer instead of sequenceNode.sugar

Reply:
Thanks, used it